### ISSUE TITLE
Moving seeds go off map

### ISSUE THREAD

**Issue by bedalton** [Creatures-Developer-Network/C2toDS-Refresh#52]
Not sure what causes it, but I have had multiple errors of pear seeds and trumpet seeds ending up off map, and throwing an error when an ATTR change is made.

The pear script has a comment addressing this, where they try to use a `MVSF` command with its current out of bounds coordinates.

I suggest killing the seeds in these situations is best. 

Initial results of killing the seed prevents the error from occurring and has not shown any negative side effects so far after 3 in game years.

**Comment by Amaikokonut** [comment]
While simply killing off-map seeds is a totally reasonable quick fix, I'd also advise looking into why the seeds are ending up off-map, with the idea that treating the root issue rather than just the symptoms can ensure that more issues don't crop up later.

**Comment by bedalton** [comment]
My best guess is they are falling through the map, but I really do not know how to diagnose the problem. When the error occurs they are way off the map. Over the errors, they pop up with an x value between ~4,000 and ~12,000, but most occur around 8,000 to 10,000. The highest y value was 23,690,526, but usually around ~50,000.

An x-value between 8,000 - 10,000 puts them over the second volcano. The four thousand I think is a fluke though, because it happened only once, everything else is over ~8700.

If you remove the highest and lowest, you have 6 points with an x-value between 8,765 and 9866. Which places them over the left 2/3rds of the second volcano.
![Fall Through](https://user-images.githubusercontent.com/3941960/125815020-12ae2047-978b-4846-86a5-e23bbd59dd38.jpg)

**Comment by bedalton** [comment]
Wait, I was wrong. I misread the map error that occurred last. It is actually negative (-9,886,-9886). It was caused by the MVSF called on the `new: simp` grass plant based on the grass seed location.

**Comment by Amaikokonut** [comment]
The fact that this seems to happen with so many seeds makes me wonder if it's a flaw inherent in how they were converted or whatever the base script used was, and that if we could find the cause in one perhaps we could apply the fix to all of them at once.

**Comment by Amaikokonut** [comment]
As a small update to this-- I've since encountered this weird error in different agents, in different metarooms (I had a lot of issues in the C3 aquatic meso), always moving to the same negative coordinates. It really is a mystery to me when and why it happens, but it _seems_ to happen only with very small agents. It might very well be an engine bug. It might be out of scope for this project but if anyone knows or wanted to investigate this on a deeper level, (@Vermidia ?) it would be a sincere relief. Buuuut meanwhile, checking if the seed is on the map before mvsf'ing it still sounds like our best bet. It does seem to happen infrequently enough, anyway.

**Comment by Amaikokonut** [comment]
Okay, Verm pointed out that the negative coordinates are actually where agents are _generated_, so the issue might actually lie in the seeds not ever being actually correctly moved from their 'spawn point' when they are initially created, possibly due to something like the script getting interrupted somewhere between the `new: simp` and the `mvto/mvsf`. So it might not be that they fall off the map so much as they _never made it onto the map in the first place_! We can check the seed creation portions of the scripts and make sure that they are run in an inst (at least between creation and the moving) and see if that prevents future issues.

I guess if that's the case it has nothing to do with the size of the agents after all! Though it makes sense that tiny agents are less likely to be missed when they don't get properly moved, too.

**Comment by Amaikokonut** [comment]
Closing this since I feel like it's been solved in a general sense but I figure if it arises again in a specific agent we can open a new issue for that specific agent.

Side note, I've been finding it useful to check which agents are doing this is with this bit of caos:
```
enum 0 0 0
	doif posx lt 0 or posy lt 0
		outv fmly
		outs " "
		outv gnus
		outs " "
		outv spcs
		outs "\n"
	endi
next
```

You get a lot of agents that run in the background and that's fine, but I figure anything with a family of 2 or higher might need to be investigated. Then you can check classifiers in the CSV in the root folder: [https://github.com/Creatures-Developer-Network/C2toDS-Refresh/blob/main/C2toDS%20Classifiers.csv](https://github.com/Creatures-Developer-Network/C2toDS-Refresh/blob/main/C2toDS%20Classifiers.csv)
